staging/src/k8s.io/apimachinery/pkg/util/json/json_test.go

```diff
@@ -417,3 +417,25 @@ func TestUnmarshalNil(t *testing.T) {
 		}
 	}
 }
+
+func TestSniff(t *testing.T) {
+	for _, tc := range []struct {
+		In      string
+		JSON    bool
+		Unknown bool
+	}{
+		{In: `                                {}`, JSON: false, Unknown: true},
```

Comment:
this is assuming json won't start with whitespace? that's true for json our encoders produce but not generally true, right?

Reply:
Here it's hitting the limit on how much of the byte slice it'll inspect. Maybe it would be more generous to guess JSON if the entire prefix is allowable JSON whitespace?